library/psa_crypto.c

```diff
@@ -3975,9 +3975,83 @@ psa_status_t psa_cipher_update( psa_cipher_operation_t *operation,
         goto exit;
     }
 
-    ret = mbedtls_cipher_update( &operation->ctx.cipher, input,
-                                 input_length, output, output_length );
-    status = mbedtls_to_psa_error( ret );
+    if( operation->alg == PSA_ALG_ECB_NO_PADDING )
+    {
```

Comment:
This is a very big block of code. Please split off either this block, or the `if( expected_output_size > 0 )` block, as a separate function.

Reply:
Done.